pkg/validations/proxy_validation.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache v2.0 license.
+
+package validations
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/microsoft/moc/pkg/errors"
+)
+
+func ValidateProxyURL(proxyURL string, certContent string) error {
+	parsedURL, err := url.ParseRequestURI(proxyURL)
+
+	if err != nil {
+		return errors.Wrapf(errors.InvalidInput, err.Error())
+	}
+
+	// Check if url scheme is http or https
+	if parsedURL.Scheme != "http" && parsedURL.Scheme != "https" {
+		return errors.Wrapf(errors.InvalidInput, "Invalid proxy URL. The URL scheme should be http or https")
+	}
+
+	// Create a certificate pool
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM([]byte(certContent))
```

Comment:
We should split this function into two, validating the URL and testing connection to the proxy

Reply:
I can rename the function as ValidateProxyUrlAndTestConection

Reply:
I thought if the function is split you can call it places where you dont need to tests the connection. So it becomes easy to from other repos

Reply:
Can you give example in which case you'll want to just validate the proxy url and not test the connection?

Reply:
This is a general coding Principle. A function is meant to perform one operation. Is there a reason we don't want to split this?

Reply:
The use cases would be when the code may want to validate the URL but may not be in the line of sight of Proxy. We are writing helper functions here and they should not tailored to particular use case.